### DISCUSSION TITLE
Migration to Hugo

### DISCUSSION THREAD

**Discussion by pourmand1376** [alshedivat/al-folio#1130] [Ideas]
I am considering creating a new repository with support for Hugo instead of Jekyll. I am satisfied with all aspects of this blog, but I am not a fan of Jekyll. I am posting this issue to gauge the level of interest in this idea and determine if it is worth further exploration.

**Comment by rohandebsarkar** [comment]
Since this is not an bug or enhancement, rather it is to see if people are interested on a port to a different framework and probably out of scope for this repository, I am converting this to a discussion.

PS: It will be very interesting to see a Hugo port of `al-folio`. I have previously thought making a port myself but held back as it might not have such a interested user base.

**Comment by rohandebsarkar** [comment]
I would very much like to see a Hugo port of `al-folio`. Thanks for proposing it here, I didn't think there might be many people interested in this.

I would very much like to collaborate to port `al-folio` to Hugo. If you are open to it, we could start working on a proof of concept to see how many people are actually interested in it.

**Reply by pourmand1376** [reply]
I would like to contribute as my Persian website is also Hugo (www.aprd.ir). 

This repo also seems interesting. We can take some ideas from it. 

https://github.com/wowchemy/starter-hugo-academic

**Comment by pourmand1376** [comment]
Useful links:
- https://gohugo.io/tools/migrations/
- https://chenhuijing.com/blog/migrating-from-jekyll-to-hugo/

**Comment by alshedivat** [comment]
excited to see you guys are considering to port al-folio to Hugo!

I've contemplated this a few years ago, as Hugo offered some nice speed improvements to the build process among other things. however, back then, Jekyll had a significantly richer ecosystem of plugins, and, as you know, pretty much all cool features of al-folio are based on various plugins. things might have changed since then, so I'm excited this pans out!

**Comment by rohandebsarkar** [comment]
After some efforts I have ported the skeleton structure of the `al-folio` theme to Hugo. Here is the repository: https://github.com/rohandebsarkar/academic-folio. And here is a preview: https://rohandebsarkar.github.io/academic-folio/

There is one major change, i.e., I have dropped Bootstrap in favor of [Tailwind CSS](https://tailwindcss.com/). I did this because Bootstrap was very heavy in size and it was quite difficult to update it to a major version (see #741).

@alshedivat dependencies is a concern for me as well, I do not yet know how to add dependencies (if they are available).

If any one wants maintain this Hugo port of `al-folio` (placeholder name `academic-folio`) in the long run just drop a comment on this post, I shall get back in touch with you.

**Reply by pourmand1376** [reply]
Thanks. The website looks great. It is clear it has taken great time and effort.

I will test the functionalities of the website and submit a PR in case I think anything is needed.

**Comment by cs-mshah** [comment]
I am highly interested in a hugo version of ai-folio with all the blog features that are there in distil blogs.
